File: src/main/rust/g0101_0200/s0128_longest_consecutive_sequence/Solution.rs
// #Medium #Top_100_Liked_Questions #Top_Interview_Questions #Array #Hash_Table #Union_Find
// #Big_O_Time_O(N_log_N)_Space_O(1) #2024_09_09_Time_4_ms_(99.44%)_Space_3.4_MB_(62.78%)

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        if nums.is_empty() {
            return 0;
        }

        let mut nums = nums;
        nums.sort_unstable(); // Equivalent to Arrays.sort in Java
        
        let mut max = i32::MIN;
        let mut ths_max = 1;

        for i in 0..nums.len() - 1 {
            if nums[i + 1] == nums[i] + 1 {
                ths_max += 1;
            } else if nums[i + 1] != nums[i] {
                max = max.max(ths_max);
                ths_max = 1;
            }
        }

        max.max(ths_max) // Return the maximum sequence length
    }
}

File: src/main/dart/g0101_0200/s0128_longest_consecutive_sequence/Solution.dart
// #Medium #Top_100_Liked_Questions #Top_Interview_Questions #Array #Hash_Table #Union_Find
// #Big_O_Time_O(N_log_N)_Space_O(1) #2024_10_07_Time_420_ms_(84.85%)_Space_163.6_MB_(90.91%)

class Solution {
  int longestConsecutive(List<int> nums) {
    if (nums.isEmpty) {
      return 0;
    }

    nums.sort(); // Sort the list
    int max = -999999999; // Equivalent to Integer.MIN_VALUE in Java
    int currentMax = 1;

    for (int i = 0; i < nums.length - 1; i++) {
      if (nums[i + 1] == nums[i] + 1) {
        currentMax += 1;
        continue;
      }
      if (nums[i + 1] == nums[i]) {
        continue;
      }
      // Start of a new sequence
      max = maxVal(max, currentMax);
      currentMax = 1;
    }

    return maxVal(max, currentMax);
  }

  int maxVal(int a, int b) => a > b ? a : b; // Utility function for max
}
